File: examples/cpp/source/quantiles/quantiles_dense_batch.cpp
/* file: quantiles_dense_batch.cpp */
/*******************************************************************************
* Copyright 2014-2018 Intel Corporation.
*
* This software and the related documents are Intel copyrighted  materials,  and
* your use of  them is  governed by the  express license  under which  they were
* provided to you (License).  Unless the License provides otherwise, you may not
* use, modify, copy, publish, distribute,  disclose or transmit this software or
* the related documents without Intel's prior written permission.
*
* This software and the related documents  are provided as  is,  with no express
* or implied  warranties,  other  than those  that are  expressly stated  in the
* License.
*******************************************************************************/

/*
!  Content:
!    C++ example of computing quantiles
!******************************************************************************/

/**
 * <a name="DAAL-EXAMPLE-CPP-QUANTILES_BATCH"></a>
 * \example quantiles_dense_batch.cpp
 */

#include "daal.h"
#include "service.h"

using namespace daal;
using namespace daal::algorithms;
using namespace daal::data_management;
using namespace std;

/* Input data set parameters */
string datasetFileName = "../data/batch/quantiles.csv";

int main(int argc, char *argv[])
{
    checkArguments(argc, argv, 1, &datasetFileName);

    /* Initialize FileDataSource<CSVFeatureManager> to retrieve the input data from a .csv file */
    FileDataSource<CSVFeatureManager> dataSource(datasetFileName, DataSource::doAllocateNumericTable, DataSource::doDictionaryFromContext);

    /* Retrieve the data from the input file */
    dataSource.loadDataBlock();

    /* Create an algorithm to compute quantiles using the default method */
    quantiles::Batch<> algorithm;

    /* Set input objects for the algorithm */
    algorithm.input.set(quantiles::data, dataSource.getNumericTable());

    /* Compute quantiles */
    algorithm.compute();

    /* Get the computed quantiles */
    quantiles::ResultPtr res = algorithm.getResult();

    printNumericTable(res->get(quantiles::quantiles), "Quantiles");

    return 0;
}

File: examples/java/com/intel/daal/examples/quantiles/QuantilesDenseBatch.java
/* file: QuantilesDenseBatch.java */
/*******************************************************************************
* Copyright 2014-2019 Intel Corporation
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*******************************************************************************/

/*
 //  Content:
 //     Java example of computing quantiles
 ////////////////////////////////////////////////////////////////////////////////
 */

/**
 * <a name="DAAL-EXAMPLE-JAVA-QUANTILESBATCH">
 * @example QuantilesDenseBatch.java
 */

package com.intel.daal.examples.quantiles;

import com.intel.daal.algorithms.quantiles.Batch;
import com.intel.daal.algorithms.quantiles.InputId;
import com.intel.daal.algorithms.quantiles.Method;
import com.intel.daal.algorithms.quantiles.Result;
import com.intel.daal.algorithms.quantiles.ResultId;
import com.intel.daal.data_management.data_source.DataSource;
import com.intel.daal.data_management.data_source.FileDataSource;
import com.intel.daal.examples.utils.Service;
import com.intel.daal.services.DaalContext;

class QuantilesDenseBatch {
    /* Input data set parameters */
    private static final String datasetFileName = "../data/batch/quantiles.csv";

    private static DaalContext context = new DaalContext();

    public static void main(String[] args) throws java.io.FileNotFoundException, java.io.IOException {

        /* Retrieve the input data */
        FileDataSource dataSource = new FileDataSource(context, datasetFileName,
                DataSource.DictionaryCreationFlag.DoDictionaryFromContext,
                DataSource.NumericTableAllocationFlag.DoAllocateNumericTable);
        dataSource.loadDataBlock();

        /* Create an algorithm to compute quantiles using the single-pass method */
        Batch algorithm = new Batch(context, Float.class, Method.defaultDense);
        algorithm.input.set(InputId.data, dataSource.getNumericTable());

        /* Compute quantiles */
        Result res = algorithm.compute();

        Service.printNumericTable("Quantiles:", res.get(ResultId.quantiles));

        context.dispose();
    }
}
